### DISCUSSION TITLE
Feedback: Retype Pro

### DISCUSSION THREAD

**Discussion by geoffreymcgill** [retypeapp/retype#224] [General]
Hello Retypers,

As of Retype `v1.11.0`, we now have all the pieces in the app in place to officially start offering a **Retype Pro** option. Before we commit to anything, we would appreciate your feedback. 

I will attempt to lay everything out here. I'm sure I'll miss some details, so if you have any questions, just let me know and I'll do my best to answer.

This post ~~may~~ will get revised as we proceed through this discussion. If details change, I will keep this first post updated to stay in sync with our current plan and feedback from you.

## History

The first commit to our **Retype App** repo was on November 25th, 2020.

<img width="525" alt="Screen Shot 2021-11-25 at 2 40 33 PM" src="https://user-images.githubusercontent.com/62210/143451819-e7ea204b-cbc8-4be6-9b20-bfc54fb6a51e.png">

So, today we celebrate the exact one-year anniversary of Retype. Let's dance!

![celebration-will-smith](https://user-images.githubusercontent.com/62210/143440691-b4a4e58e-3c47-4814-931b-0e6aa8b80ac6.gif)

We started Retype because we needed the tool. We wanted something like Retype, although there were just no other tools, projects, or services available that met our requirements. 

We were also more than happy to pay for something if it checked at least most of the boxes. A couple of products and projects came close but there were always compromises and ultimately deal breakers.

The most ideal number of software projects available is always `n + 1` 🙃 , so of course like any foolhardy software developer we said _"Let's just build it"_.

![fieldofdreams-ifyoubuildit](https://user-images.githubusercontent.com/62210/143443767-c306a44b-2c50-456f-a965-d74e9c5b9cc8.gif)

Writing software and the expenses of a small team do not come cheap. To date, the equivalent of at least three [kick ass] engineers have worked full time on Retype for one year. You do the math on what that might cost.

The project has made $0 in sales and has taken $0 in investment. Retype has been 100% bootstrapped and our preference is to keep it that way.

Our goal is to build something useful, challenge ourselves, have fun, and maybe MAYBE we could earn enough revenue to pay our ongoing costs. Even better, grow a little. We could, and should, hire 2-4 people today to have a fighting chance of building the current vision within a reasonable amount of time.

There's sooo much more we could be doing, but this shit takes time and money.

Here's our proposal...

## The Proposal

Offer a **Retype Pro** build option with additional features and services that some users may find value in.

Current **Retype** vs **Retype Pro**  comparison matrix:

Feature   | Retype | Retype Pro
---       | ---    | ---
Pricing   | FREE   | ~~$149~~ $99 USD/yr for Early Adopters
Users     | Unlimited | Unlimited
Renewal   | It's FREE of course | $99 USD for another year
Version upgrades | All | All, up until key expires
Max pages (per project) | 100    | 1000
Max websites | Unlimited | 5
Option to remove **Powered by Retype** branding | No | Yes
Future features | Yes | Yes, plus extended functionality

### Q. Does this mean the free Retype is going away?

The free version of Retype is here to stay and is being constantly updated with new features and fixes.

**Retype Pro** just unlocks additional or extended functionality, such as building a project with more than 100 `.md` pages and a config to remove the **Powered by Retype** branding.

### Q. Will **Retype Pro** be a separate product that I need to install?

No. The key will unlock functionality within the same `retypeapp` that you have already installed. No separate installer or product installation is required.

Just add the key to your project and you are good to go. 

### Q. If I do not renew, does my website stop working?

Your website will continue to work as it has always worked.

The only thing that changes is that your project will not be able to upgrade to the next Retype release.

### Q. What is considered a website?

We consider a website a unique URL. 

Sub-domains each count as one unique URL. 

Each key will unlock up to 5 unique URLs.

As an example, with a valid key, projects with the following URLs will be possible to build as **Retype Pro**:

URL | Included?
--- | ---
`example.com` | :white_check_mark:
`docs.example.com` | :white_check_mark:
`staging.example.com` | :white_check_mark:
`docs.projectx.com` | :white_check_mark:
`github.com/user123/my-project` | :white_check_mark:
`my-next-project.io` | :no_entry_sign:

The URL is baked right into the key. You can have any number of projects running on any number of developer machines using the same key, but the key will be limited to building only those **Retype Pro** projects with a `retype.yml` file where the `url` is set with one of your defined website URLs.

### Q. Can we purchase a multi-year key and is a discount available?

Purchase for one year, or for as long as you want. 

If you purchase 2 or more years, take an additional 10% off the total.

For instance, purchasing 2 years would be `($149 + $99) - 10% = $232.20 USD`, although discounted pricing for Early Adopters would be `($99 + $99) - 10% = $178.20. USD`.

Think you will be using Retype for 5 years? Pricing as an Early Adopter would be `($99 * 5) - 10% = $445.50 USD`.

You are running 8 websites with Retype and have a 4-year time frame. Pricing today would be `($99 * 2 * 4) - 10% = $712.80 USD`.

### Q. What if I have more than 5 projects?

No problem. Just purchase another Retype Pro instance. Need keys for up to 10 websites? Purchase two instances of **Retype Pro**.

We are building a portal where you will be able to self-manage your keys. Purchase, renew, add, remove, etc. 

### Q. We have 3 people on our team writing content and using Retype. Do we require 3 licenses?

No. Retype is licensed per website, not per person. One **Retype Pro** instance will cover an unlimited number of content authors or developers using Retype to build up to 5 unique websites.

### Q. What happens when the key expires?

If you do not renew for another term, your project stays on the last release available as of your expiry date. You will not be able to upgrade your project to the next release of Retype.

Let's run through two scenarios based on purchasing **Retype Pro** today (2021-11-25) using the following details:

- You purchase today (2021-11-25)
- Current release is Retype `v1.11.0`
- Key expires on 2022-11-25
- On the key expiry date (2022-11-25), let's say Retype `v1.20.0` is the current release

#### Scenario 1: Renew

You like Retype. It is bringing you value and happiness. 🥰  You want to renew. After purchasing a renewal, you will get another key with a new expiry date. Add that new key to your project and you are unlocked until that key expires. All upgrades are included while the key is valid.

#### Scenario 2: No renew

You decide NOT to renew your Retype key. No worries. Your project will continue to build forever, except you cannot upgrade to the next Retype release. You would stay on the current release.

You would be able to upgrade Retype up until `v1.20.0`. 

Once `v1.21.0` is released, your project would fail to build if you upgraded. You would have to downgrade back to the `v1.20.0` release.

You would still be able to build up to 1000 pages too and remove the **Powered by Retype** branding. Again, you just would not be able to upgrade to any future releases.

If at some point you would like to upgrade to the latest release, just purchase a new **Retype Pro** key at that time.

---

At the moment, we do not have an option to purchase **Retype Pro**. The above is a proposal. If the general consensus from the community is to go ahead, then we will open up an online purchase option based on the approved pricing and plan at that time.

If you want to use **Retype Pro** today and are generally happy with the proposal, please send me email (hello@retype.com) and I will send back a temporary 30-day key to unlock the **Retype Pro** build options.

We would like to leave this proposal open for feedback and modification for up to one week. Decisions need to be made by December 1, 2021.

The above details are not chiseled in stone. Details are subject to change. If you have questions, comments, or feedback, please let us know down below.

Thoughts?

**Comment by DannyBen** [comment]
First of all, congratulations on the project anniversary, and for reaching the Pro era.

My initial thoughts are these:

1. I feel that the proposed limitations of the free account are generous, which is great. 
2. Personally, I do not like subscription models for software. It feels like a model that was invented by sales people, not product people. I can understand it when it pays for ongoing things (e.g. hosting), not for software. For software, I feel that I should be paying for something that _was_ developed, not for something that _will be_ developed.
3. I like the pricing model [presented by Sublime Text][1]: You pay once, it gets you 3 years of updates, and afterwards you can continue to use your older version or pay again if you want upgrades.

[1]: https://www.sublimehq.com/store/text

**Reply by geoffreymcgill** [reply]
Thanks for the feedback.

> Personally, I do not like subscription models for software. It feels like a model that was invented by sales people, not product people. I can understand it when it pays for ongoing things (e.g. hosting), not for software. For software, I feel that I should be paying for something that was developed, not for something that will be developed.

Agreed. We took a lot of inspiration from the Sublime Text licensing model, although wanted to be even more generous with our offerings. Hence why Retype will always work and is not a trial or time limited product, unlimited users for Retype Pro vs per seat with Sublime, etc.

The subscription model for Retype was avoided, just like Sublime, although I guess the primary difference I see is 3 years of free upgrades for Sublime vs 1 year of free upgrades for Retype. You can also get three years of upgrades with Retype if required.

Both Sublime and the Retype Pro would continue to work _"as is"_ after that initial upgrade period. If beyond, then a renewal is required.

There will be a difference between Sublime and Retype once the new **Retype Pro** services are available. Those services will have a hosting cost for us. They will just be included and magically appear for Retype Pro builds, but for us, they will have infrastructure costs. Full-text search, comments, and likes are all in the planning stages. They are not a feature of Retype Pro right now, so any purchase should be made with the understanding that those features may never be available. If/when they do become available, they will be included in the free upgrades of Retype Pro. Sublime does not offer additional services (that I am aware of), so their installed product is their product and has no runtime cost to Sublime.

Just to clarify and to reiterate from the proposal above, you are not _subscribing_ to Retype. You are purchasing rights to use the software (and services) and will receive free upgrades for at least one year. At the end of the term if you do not wish to upgrade, you stay on the latest version of Retype released at that time, just like Sublime. Your project can stay on that release forever.

If you do not require any further upgrades, you just seamlessly continue with your current setup.

Hope this helps clarify.

**Reply by DannyBen** [reply]
> Just to clarify and to reiterate from the proposal above, you are not _subscribing_ to Retype. You are purchasing rights to use the software (and services) and will receive free upgrades for at least one year. At the end of the term if you do not wish to upgrade, you stay on the latest version of Retype released at that time, just like Sublime. Your project can stay on that release forever.

So yes, this wasn't clear to me. So it means that if someone is purchasing because they just have 101 pages, they will still be able to show all 101 pages after their license has expired. Right? If so, this is better indeed.

Now that you are talking about services (i.e. things that _do_ incur ongoing cost on Retype the company), I feel that there will be no choice in the future to provide different payment tiers, for different use cases. Since page limit and hosted search are two very different needs.

**Reply by geoffreymcgill** [reply]
> So it means that if someone is purchasing because they just have 101 pages, they will still be able to show all 101 pages after their license has expired. Right? If so, this is better indeed.

Exactly. Nothing will change with their build process or their version of Retype. Everything will continue to function as normal. Even adding more pages (up to 1000) would be just fine.

The only difference being the project will not be able to upgrade to the next Retype release. Or, they just move to the non-pro feature set of Retype. 

> Now that you are talking about services (i.e. things that do incur ongoing cost on Retype the company), I feel that there will be no choice in the future to provide different payment tiers, for different use cases. Since page limit and hosted search are two very different needs.

The goal is to not scale pricing. There are still some questions still to be answered, but we're going to try and make it work. The system we are planning will be extremely efficient, so that will help optimize the server resources and keep our operational cost down to a minimum. 

_Maybe_ some well documented upper limits would be required, but I feel a site would need to be at least a couple stdev away on the high side to hit those.

**Comment by andrew-culver** [comment]
Congrats on the anniversary. 

One gap I see is how you will cover the cost of continuing hosted services for a Pro account that is not renewed.

The rest of the deal seems well considered.

**Reply by geoffreymcgill** [reply]
> One gap I see is how you will cover the cost of continuing hosted services for a Pro account that is not renewed.

Yes, agreed. We have some solid ideas on how to manage. No matter what, if you choose to not renew/extend a Retype Pro key, your website(s) and features such as full-index search will still continue to work.

**Reply by geoffreymcgill** [reply]
Here is one compromise we may have to settle on for Retype Pro projects that do not renew but are still consuming Retype Pro Services. The option to remove the `Powered by Retype` branding may be disabled on the next expired build. If you had the `Powered by Retype` branding hidden, that `retype.yml` option would be ignored on the next build and the branding would be rendered. 

The current plan is still to keep everything else running the same. Seems like a fair compromise.

**Reply by DannyBen** [reply]
I feel this is more than fair, it is generous.

**Reply by andrew-culver** [reply]
Rendering the Retype branding again upon paid plan cancellation is totally reasonable, I would say even expected.

I even think it is reasonable to cancel the other paid features. Otherwise, you are motivating customers to game you for free extra services, while customers who continue to pay are not rewarded for not gaming you.

I plan to pay you. I plan to not take advantage of paying then cancelling. What do I get for my honesty (other than the removal of your branding, which ain't much)?

**Reply by geoffreymcgill** [reply]
> What do I get for my honesty (other than the removal of your branding, which ain't much)?

You get to upgrade and enjoy all the wonderful new features and bug fixes. New features and fixes that will not be available to non-pro sites.

It will be up to Retype to deliver continuous value. If you see no new value, then you have the option to full stop or at least delay upgrading for a while.

The cost to Retype of delivering those services is an excellent motivator for us to build and operate those services as efficiently as possible, which in turn benefits Pro sites. The `Powered by Retype` branding is a small chunk of quid pro quo, which we feel is a fair trade.

**Comment by andrew-culver** [comment]
What's the latest on Pro? When can I pay? ;-)

**Comment by ApolonTorq** [comment]
Just saw this post. Some questions include:

- API docs can contain a heap of auto-generated pages from source code. Are they included in the 1000 page count per project?
- Do deployment pipeline (manual or automated) sites all contribute to the project count? e.g. a localhost url individual work, a LAN url for review within an organization, a test site url , a staging before production url and then a public production url. Not all of these examples are needed in all instances, but it illustrates the intent of the question with respect to unique urls. Given the price, its not a big deal. However it may be worth considering exceptions for localhost and IP address urls? i.e. something that distinguishes between deploying the final result and builds on the road to the final result. One approach could be a banner on every page indicating “Test”, “Pre-Release” or similar and have those not contribute to the project license counts.
- Can a licensee mix and match Retype licensing for different projects? e.g. one project that doesn’t need Pro features uses the free version but another project that needs the Pro features uses the commercial license.

**Comment by ddnexus** [comment]
### Suggestions:

1. Open source projects should have an OS license that should allow all the features unlocked.
2. All the project contributors should be able to use it fully, obviously strictly limited to the project repo/site.

**Comment by itsthejoker** [comment]
I love the idea of a Pro account for companies, but I would also like to propose a one-time open source "I need more than the free tier" option if @ddnexus' suggestion of a general "open source gets pro for free" idea is too much. As an example, the project that I'm currently deploying Retype for is pretty large and could easily balloon over 100 pages, but I can't afford to put $100 a year just into documentation. I'd be happy to give you $50 or $100 as a one-time "we're complicated but also thanks for cutting us a break since we're open source" fee.

**Reply by ddnexus** [reply]
I don't think it's too much.

For example JetBrains gives you the [OS License](https://www.jetbrains.com/community/opensource/#support) for ALL their softwares together (and they cover really more than you will ever need) if you have a decent OS project. Their limits are pretty generous, and commercially it's a good move for their business.

The "open source gets pro for free" principle is very ethical and should be the default for all commercial projects that use themselves OS software for making money!

**Reply by geoffreymcgill** [reply]
>  I can't afford to put $100 a year just into documentation

Just to clarify, the $99 renewal as made in the proposal is 100% optional. If you did not renew, your project would still be able to happily use the last release available before the expiry of the license.

**Comment by geoffreymcgill** [comment]
Purchasing Retype Pro is now available at https://retype.com/pro.

We moved to include three URLs instead of the proposed five, but also included any projects using your `github.io` or `gitlab.io` organization under one domain, so building more than three Retype Pro projects is possible. See https://retype.com/pro/#what-is-considered-a-website

We are working on a new system to enable self-management of license keys.

New form for creating a trial key is online too:
https://retype.com/pro/#can-i-get-a-trial-license-key

Hope this helps.

**Reply by geoffreymcgill** [reply]
[![IMG_960CB9796826-1](https://user-images.githubusercontent.com/62210/161127078-3697f8b3-3f87-4196-87d6-61669b55bbcc.jpeg)](https://twitter.com/retypeapp/status/1509601130610790402)
